### PR TITLE
backport: changes from master to develop

### PR DESCRIPTION
**Pull request by tamtamchik**
Want to prepare the develop branch to be ported to lido-dao.

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -7,6 +7,11 @@ jobs:
     name: Hardhat
     runs-on: ubuntu-latest
 
+    permissions:
```
Comment (rnmsslido) [inline .github/workflows/coverage.yml:10]: Now I see these permissions are maybe too open; have you tested it with only `contents: read`?
If no actions are using PRs and issues, it should be enough.

Comment (tamtamchik) [reply .github/workflows/coverage.yml:10]: Changed in 17c5ca7390d3038287bb3d9d7424aadac0922e3f

**Review by TheDZhon — APPROVED** [review]
image

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (tamtamchik) [reply .github/workflows/coverage.yml:10]: Well, it pushes to the coverage reports branch (to be able to make diffs), so it needs the write permissions.

**Comment by TheDZhon** [comment]
image

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (rnmsslido) [reply .github/workflows/coverage.yml:10]: I see, in this case `contents: write` is needed indeed. These ones could be removed, then?
```yaml
      issues: write
      pull-requests: write
 ```
Should be a good test to do on `develop`.

**Pull request merged**